wemake_python_styleguide/visitors/ast/blocks.py

```diff
@@ -161,11 +161,11 @@ def _scope(
     ) -> None:
         scope = BlockScope(node)
         shadow = scope.shadowing(names, is_local=is_local)
-
         if shadow:
-            self.add_violation(
-                BlockAndLocalOverlapViolation(node, text=', '.join(shadow)),
-            )
+            if not is_imported_var_assigned(node):
```

Comment:
Sorry, I was not clear enough in my task description. As a result you took a wrong path. Let me explain the task idea once again.

Imagine that we have a block variable `my_var`. We don't care about how exactly it is defined. We only care that it is the block one.

We also have a local variable `my_var` that shadows the block one. Normally, we would raise a violation. But, in case if local `my_var` definition uses the block one - it is okay.

Example:

```python
for my_var in some_attrs():
    my_var = my_var.lower()  # ok
```

Reply:
@sobolevn do I disable this rule for cases with any block variables?

Reply:
Sorry, I don't get your question. You can use Russian in case it is easier for you.

Reply:
Переопределение в любом блоке должно быть разрешено для случаев, когда новое значение использует старое?

Reply:
Yeap 👍